orchestrationSpecs/userConfigSchemas.txt

```diff
@@ -0,0 +1,59 @@
+HTTP_AUTH_BASIC: {
+  username: string
+  password: string
+}
+
+HTTP_AUTH_SIGV4: {
+  region:  string
+  service?: string | "es" // default
+}
+
+CLUSTER_CONFIG: {
+  name:            string
+  endpoint?:       string // optional for source and maybe metadata evaluate commands, but required for most others
+  allow_insecure?: bool
+  version?:        string // required in some contexts?
+  authConfig?:     HTTP_AUTH_BASIC | HTTP_AUTH_SIGV4
+}
+
+TARGET_CLUSTER_CONFIG: CLUSTER_CONFIG {
+  endpoint: string
+}
+
+SNAPSHOT_MIGRATION_CONFIG: {
+	indices: [string]
+	migrations: [{
+		metadata: {
+			mappings: {
+				properties: {...}
+			},
+			//
+			documentBackfillConfigs: {
+				indices: [...string],
```

Comment:
t seems like we will need to have more complex validation on this code where as a model expected multiple snapshot migration configs would be less error prone.

Consider these questions, why is there an outer indices and inner indices object?  Is this about allocating the indices to specific backfill parameters then it is possible some are included multiple times or accidentally not included?